src/services/db/GitFileSystemService.ts

```diff
@@ -0,0 +1,278 @@
+import fs from "fs"
+
+import { errAsync, okAsync, Result } from "neverthrow"
+import { GitError, SimpleGit } from "simple-git"
+
+import { config } from "@config/config"
+
+import GitFileSystemError from "@errors/GitFileSystemError"
+
+import { ISOMER_GITHUB_ORG_NAME } from "@constants/constants"
+
+import type { GitDirectoryItem, GitFile } from "@root/types/gitfilesystem"
+
+/**
+ * Some notes:
+ * - Seems like getTree, updateTree and updateRepoState is always used together
+ */
+
+const EFS_VOL_PATH = config.get("aws.efs.volPath")
+
+export default class GitFileSystemService {
+  private readonly git: SimpleGit
+
+  constructor(git: SimpleGit) {
+    this.git = git
+  }
+
+  // Determine if the folder is a valid Git repository
+  async isValidGitRepo(
+    repoName: string
+  ): Promise<Result<boolean, GitFileSystemError | unknown>> {
+    const originUrl = `git@github.com:${ISOMER_GITHUB_ORG_NAME}/${repoName}.git`
+
+    try {
+      // Check if an existing folder exists
+      if (!fs.existsSync(`${EFS_VOL_PATH}/${repoName}`)) {
+        return okAsync(false)
+      }
+
+      // Check if the folder is a Git repo
+      await this.git.cwd(`${EFS_VOL_PATH}/${repoName}`)
+      const isGitRepo = await this.git.checkIsRepo()
+
+      if (!isGitRepo) {
+        return okAsync(false)
+      }
+
+      // Check if the Git repo is the correct one
+      const remoteUrl = await this.git.remote(["get-url", "origin"])
+      if (!remoteUrl || remoteUrl.trim() !== originUrl) {
+        return okAsync(false)
+      }
+
+      return okAsync(true)
```

Comment:
Hmm for this case, the `false` is expected because the function is supposed to return a boolean type. I kept the `errAsync` for only instances where an actual error occurred.

Reply:
actually don't really have to but since it's here alr it's fine!

Reply:
I think I misunderstood how neverthrow works and kish walked me through it just now, done in f9fb3b337cd02d66c2c05eb50f6e88b389196d9b.